core/types/receipt.go

```diff
@@ -392,20 +400,14 @@ func (r Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, num
 		r[i].TransactionIndex = uint(i)
 
 		// The contract address can be derived from the transaction itself
-		if r[i].ContractAddress.Equal(common.Address{}) && txs[i].To() == nil {
+		if txs[i].Type() == QuaiTxType && r[i].ContractAddress.Equal(common.Address{}) && txs[i].To() == nil {
 			// Deriving the signer is expensive, only do if it's actually needed
 			from, err := Sender(signer, txs[i])
 			if err != nil {
 				return err
 			}
 			r[i].ContractAddress = crypto.CreateAddress(from, txs[i].Nonce(), txs[i].Data(), config.Location)
 		}
-		// The used gas can be calculated based on previous r
-		if i == 0 {
-			r[i].GasUsed = r[i].CumulativeGasUsed
```

Comment:
we need this field right? why did we remove?

Reply:
This doesn't need to be calculated, in fact it's wrong currently. All receipts have a GasUsed field